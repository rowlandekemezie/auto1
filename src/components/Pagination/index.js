import React from "react";
import styled, { css } from "styled-components";
import PropTypes from "prop-types";

const Wrapper = styled("div")`
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 24px;
`;

const Text = styled("span")`
  color: ${({ theme }) => theme.LIGHT_ORANGE};
  font-size: 12px;
  margin: 0 24px;
  cursor: pointer;

  ${props =>
    props.disabled
      ? css`
          color: ${({ theme }) => theme.BLACK};
          cursor: not-allowed;
        `
      : null}
`;

export default function Pagination({
  onNext,
  onPrevious,
  totalPageCount,
  currentPage,
  previousPage
}) {
  console.log(totalPageCount);

  return (
    <Wrapper>
      <Text onClick={() => onNext(1)} disabled={currentPage === 1}>
        First
      </Text>
      <Text
        onClick={() => onPrevious(previousPage)}
        disabled={previousPage === currentPage}
      >
        Previous
      </Text>
      <Text disabled={true}>
        Page {currentPage} of {totalPageCount}{" "}
      </Text>
      <Text
        onClick={() => onNext(currentPage + 1)}
        disabled={currentPage === totalPageCount}
      >
        Next
      </Text>
      <Text
        onClick={() => onNext(totalPageCount)}
        disabled={currentPage === totalPageCount}
      >
        Last
      </Text>
    </Wrapper>
  );
}

Pagination.propTypes = {
  onNext: PropTypes.func.isRequired,
  onPrevious: PropTypes.func.isRequired,
  totalPageCount: PropTypes.number,
  currentPage: PropTypes.number,
  previousPage: PropTypes.number
};
