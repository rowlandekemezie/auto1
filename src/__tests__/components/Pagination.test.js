import React from "react";
import { shallow, mount } from "enzyme";
import Pagination from "../../components/Pagination";
import { SSL_OP_ALLOW_UNSAFE_LEGACY_RENEGOTIATION } from "constants";

describe("<Pagination />", () => {
  const props = {
    onNext: jest.fn(),
    onPrevious: jest.fn(),
    currentPage: 1,
    totalPageCount: 2
  };
  it("should render component properly", () => {
    const wrapper = shallow(<Pagination {...props} />);
    expect(wrapper.length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });

  it("should props with the required proptypes", () => {
    const wrapper = mount(<Pagination {...props} />);
    expect(wrapper.props().onNext).toBeFunction();
    expect(wrapper.props().onPrevious).toBeFunction();
    expect(wrapper.props().currentPage).toBeNumber();
    expect(wrapper.props().totalPageCount).toBeNumber();
  });

  it("should show disabled text and updated properly", () => {
    const wrapper = mount(<Pagination {...props} />);

    const spans = wrapper.find("span");
    expect(spans.length).toBe(5);
    expect(spans.at(0).props().disabled).toBeTruthy();
    expect(spans.at(2).props().disabled).toBeTruthy();
    expect(spans.at(4).props().disabled).toBeFalsy();
    expect(spans.at(2).text()).toEqual(
      `Page ${props.currentPage} of ${props.totalPageCount} `
    );
    expect(spans.at(4).text()).toEqual(`Last`);
  });
});
