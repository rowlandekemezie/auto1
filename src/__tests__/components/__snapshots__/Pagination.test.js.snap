// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> should render component properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Pagination
    currentPage={1}
    onNext={[MockFunction]}
    onPrevious={[MockFunction]}
    totalPageCount={2}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef
          disabled={true}
          onClick={[Function]}
        >
          First
        </ForwardRef>,
        <ForwardRef
          disabled={false}
          onClick={[Function]}
        >
          Previous
        </ForwardRef>,
        <ForwardRef
          disabled={true}
        >
          Page 
          1
           of 
          2
           
        </ForwardRef>,
        <ForwardRef
          disabled={false}
          onClick={[Function]}
        >
          Next
        </ForwardRef>,
        <ForwardRef
          disabled={false}
          onClick={[Function]}
        >
          Last
        </ForwardRef>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "First",
          "disabled": true,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "First",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "rules": Array [
              "
  color: ",
              [Function],
              ";
  font-size: 12px;
  margin: 0 24px;
  cursor: pointer;

  ",
              [Function],
              "
",
            ],
          },
          "displayName": "styled.span",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "span",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Previous",
          "disabled": false,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Previous",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "rules": Array [
              "
  color: ",
              [Function],
              ";
  font-size: 12px;
  margin: 0 24px;
  cursor: pointer;

  ",
              [Function],
              "
",
            ],
          },
          "displayName": "styled.span",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "span",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            "Page ",
            1,
            " of ",
            2,
            " ",
          ],
          "disabled": true,
        },
        "ref": null,
        "rendered": Array [
          "Page ",
          1,
          " of ",
          2,
          " ",
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "rules": Array [
              "
  color: ",
              [Function],
              ";
  font-size: 12px;
  margin: 0 24px;
  cursor: pointer;

  ",
              [Function],
              "
",
            ],
          },
          "displayName": "styled.span",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "span",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Next",
          "disabled": false,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Next",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "rules": Array [
              "
  color: ",
              [Function],
              ";
  font-size: 12px;
  margin: 0 24px;
  cursor: pointer;

  ",
              [Function],
              "
",
            ],
          },
          "displayName": "styled.span",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "span",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Last",
          "disabled": false,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Last",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "rules": Array [
              "
  color: ",
              [Function],
              ";
  font-size: 12px;
  margin: 0 24px;
  cursor: pointer;

  ",
              [Function],
              "
",
            ],
          },
          "displayName": "styled.span",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "span",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 24px;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef
            disabled={true}
            onClick={[Function]}
          >
            First
          </ForwardRef>,
          <ForwardRef
            disabled={false}
            onClick={[Function]}
          >
            Previous
          </ForwardRef>,
          <ForwardRef
            disabled={true}
          >
            Page 
            1
             of 
            2
             
          </ForwardRef>,
          <ForwardRef
            disabled={false}
            onClick={[Function]}
          >
            Next
          </ForwardRef>,
          <ForwardRef
            disabled={false}
            onClick={[Function]}
          >
            Last
          </ForwardRef>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "First",
            "disabled": true,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "First",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": false,
              "rules": Array [
                "
  color: ",
                [Function],
                ";
  font-size: 12px;
  margin: 0 24px;
  cursor: pointer;

  ",
                [Function],
                "
",
              ],
            },
            "displayName": "styled.span",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "span",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Previous",
            "disabled": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Previous",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": false,
              "rules": Array [
                "
  color: ",
                [Function],
                ";
  font-size: 12px;
  margin: 0 24px;
  cursor: pointer;

  ",
                [Function],
                "
",
              ],
            },
            "displayName": "styled.span",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "span",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              "Page ",
              1,
              " of ",
              2,
              " ",
            ],
            "disabled": true,
          },
          "ref": null,
          "rendered": Array [
            "Page ",
            1,
            " of ",
            2,
            " ",
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": false,
              "rules": Array [
                "
  color: ",
                [Function],
                ";
  font-size: 12px;
  margin: 0 24px;
  cursor: pointer;

  ",
                [Function],
                "
",
              ],
            },
            "displayName": "styled.span",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "span",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Next",
            "disabled": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Next",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": false,
              "rules": Array [
                "
  color: ",
                [Function],
                ";
  font-size: 12px;
  margin: 0 24px;
  cursor: pointer;

  ",
                [Function],
                "
",
              ],
            },
            "displayName": "styled.span",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "span",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Last",
            "disabled": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Last",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": false,
              "rules": Array [
                "
  color: ",
                [Function],
                ";
  font-size: 12px;
  margin: 0 24px;
  cursor: pointer;

  ",
                [Function],
                "
",
              ],
            },
            "displayName": "styled.span",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "span",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 24px;
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
